core/sr-io/without_std.rs

```diff
@@ -109,6 +116,19 @@ pub fn storage(key: &[u8]) -> Option<Vec<u8>> {
 	}
 }
 
+/// Get `key` from child storage and return a `Vec`, empty if there's a problem.
+pub fn child_storage(storage_key: &[u8], key: &[u8]) -> Option<Vec<u8>> {
+	let mut length: u32 = 0;
+	unsafe {
+		let ptr = ext_get_allocated_child_storage(storage_key.as_ptr(), storage_key.len() as u32, key.as_ptr(), key.len() as u32, &mut length);
+		if length == u32::max_value() {
+			None
+		} else {
+			Some(Vec::from_raw_parts(ptr, length as usize, length as usize))
```

Comment:
> free ptr

Do you have an idea how exactly?

Reply:
I think this is fine. This is also what we use for `storage(..)` on L114. We create the pointer from `ext_get_allocated_storage` or `ext_get_allocated_child_storage`, and `length` is always modified to be what's required.

Reply:
Ah okay I get what you mean here. It looks like we're already leaking heap memory for existing operations. I don't think I have good idea to fix that in this PR, but I'll create a new issue.

Reply:
https://github.com/paritytech/substrate/issues/927

Reply:
For now it should be probably ok to allocate memory in the host and deallocate it via `Vec::drop` in the runtime. But we will need to address this before #927 / #300 since when we actually do change the allocator this will transform into UB.

Reply:
Hmm okay, does not sound that nice :D

Reply:
@pepyakin @bkchr I changed those to use `slice::from_raw_parts(..).to_vec()`.